gateway/coprocess.go

```diff
@@ -362,11 +362,15 @@ func (m *CoProcessMiddleware) ProcessRequest(w http.ResponseWriter, r *http.Requ
 		if returnObject.Session == nil || token == "" {
 			authHeaderValue := r.Header.Get(m.Spec.Auth.AuthHeaderName)
 			AuthFailed(m, r, authHeaderValue)
-			return errors.New("Key not authorised"), 403
+			return errors.New(http.StatusText(http.StatusForbidden)), http.StatusForbidden
 		}
 
 		returnedSession := TykSessionState(returnObject.Session)
 
+		if err := m.ApplyPolicies(returnedSession); err != nil {
```

Comment:
shouldn't this be before TykSessionState?

Reply:
@letzya Actually before line 368, we don't know the contents of the session. So it's ok to call `ApplyPolicies` after it.